tket/src/Gate/Gate.cpp

```diff
@@ -437,6 +440,14 @@ std::vector<Expr> Gate::get_tk1_angles() const {
           params_.at(1) + 0.5, params_.at(0), params_.at(2) - 0.5,
           (params_.at(1) + params_.at(2)) / 2};
     }
+    case OpType::NPhasedX: {
+      if (n_qubits_ != 1) {
+        throw NotValid(
```

Comment:
Do we need this condition? Or can we have it decompose to N TK1?

Reply:
It seems to be needed. This function returns 4 expressions (3 angles plus phase) for the TK1; it is only designed to work for single-qubit gates.